mailpoet/assets/js/src/newsletters/send/filter_segment.tsx

```diff
@@ -0,0 +1,147 @@
+import { NewsLetter } from 'common/newsletter';
+import { Field } from 'form/types';
+import { ChangeEvent, useCallback, useEffect, useMemo, useState } from 'react';
+import { __ } from '@wordpress/i18n';
+import { Selection } from 'form/fields/selection';
+import { Toggle } from 'common';
+import { Tooltip } from 'common/tooltip/tooltip';
+import { Icon, help } from '@wordpress/icons';
+
+type FilterSegmentProps = {
+  item?: NewsLetter;
+  onValueChange: (targetWrap: {
+    target: {
+      name: string;
+      value: unknown;
+    };
+  }) => void;
+  field: Field;
+};
+
+export function FilterSegment({
+  item,
+  onValueChange,
+  field,
+}: FilterSegmentProps) {
+  const currentFilterSegmentId = item?.options.filterSegmentId;
+  const premiumFeaturesAvailable: boolean = useMemo(
```

Comment:
I'm asking this more to learn React than anything else, but why do we need `useMemo()` here? My limited understanding is that it should be used for performance optimization when whatever is calculated inside it is an expensive calculation, which doesn't seem to be the case here, but I'm probably missing something.

My concern without checking the details is that this could add more overhead than benefits. I don't know if that is the case here or not.

Reply:
Thank you for bringing this up! You're right, it's not necessary and probably does overcomplicate things unnecessarily. Sometimes I reach for hooks reflexively and this was one of those cases.

In any case, this is simplified now that I'm pulling in the get `premiumFeaturesAvailable` value.